Script with id send_packet causes bootloop

### The problem

Setting script id to send_packet causes bootloop

### Which version of ESPHome has the issue?

ESPHome version 2025.9.3

### What type of installation are you using?

Docker

### What platform are you using?

ESP32

### Component causing the issue

IDF UART / script

### YAML Config

```yaml
esphome:
  name: esphome-web-ecf6d8
  friendly_name: esphome-test-32s3
  min_version: 2025.9.0
  name_add_mac_suffix: false

esp32:
  variant: esp32s3
  framework:
    type: esp-idf

# Enable logging
logger:
  hardware_uart: UART0
  level: DEBUG

# Enable Home Assistant API
api:

# Allow Over-The-Air updates
ota:
- platform: esphome

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

script:
  - id: send_packet
    then:
```

### Anything in the logs that might be useful for us?

```txt
[21:49:07]ESP-ROM:esp32s3-20210327
[21:49:07]Build:Mar 27 2021
[21:49:07]rst:0xc (RTC_SW_CPU_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
[21:49:07]Saved PC:0x40376779
[21:49:07]SPIWP:0xee
[21:49:07]mode:DIO, clock div:1
[21:49:07]load:0x3fce2810,len:0x1564
[21:49:07]load:0x403c8700,len:0x4
[21:49:07]load:0x403c8704,len:0xd24
[21:49:07]load:0x403cb700,len:0x2ed4
[21:49:07]entry 0x403c8920
[21:49:07]I (31) boot: ESP-IDF 5.4.2 2nd stage bootloader
[21:49:07]I (31) boot: compile time Oct  2 2025 18:45:39
[21:49:07]I (31) boot: Multicore bootloader
[21:49:07]I (31) boot: chip revision: v0.2
[21:49:07]I (34) boot: efuse block revision: v1.3
[21:49:07]I (38) boot.esp32s3: Boot SPI Speed : 80MHz
[21:49:07]I (42) boot.esp32s3: SPI Mode       : DIO
[21:49:07]I (45) boot.esp32s3: SPI Flash Size : 4MB
[21:49:07]I (49) boot: Enabling RNG early entropy source...
[21:49:07]I (54) boot: Partition Table:
[21:49:07]I (56) boot: ## Label            Usage          Type ST Offset   Length
[21:49:07]I (63) boot:  0 otadata          OTA data         01 00 00009000 00002000
[21:49:07]I (69) boot:  1 phy_init         RF data          01 01 0000b000 00001000
[21:49:07]I (76) boot:  2 app0             OTA app          00 10 00010000 001c0000
[21:49:07]I (82) boot:  3 app1             OTA app          00 11 001d0000 001c0000
[21:49:07]I (89) boot:  4 nvs              WiFi data        01 02 00390000 0006d000
[21:49:07]I (95) boot: End of partition table
[21:49:07]I (98) esp_image: segment 0: paddr=00010020 vaddr=3c090020 size=17f88h ( 98184) map
[21:49:07]I (123) esp_image: segment 1: paddr=00027fb0 vaddr=3fc98d00 size=0486ch ( 18540) load
[21:49:07]I (127) esp_image: segment 2: paddr=0002c824 vaddr=40374000 size=037f4h ( 14324) load
[21:49:07]I (131) esp_image: segment 3: paddr=00030020 vaddr=42000020 size=80a20h (526880) map
[21:49:07]I (227) esp_image: segment 4: paddr=000b0a48 vaddr=403777f4 size=11424h ( 70692) load
[21:49:07]I (243) esp_image: segment 5: paddr=000c1e74 vaddr=600fe000 size=0001ch (    28) load
[21:49:07]I (251) boot: Loaded app from partition at offset 0x10000
[21:49:07]I (252) boot: Disabling RNG early entropy source...
[21:49:07][I][logger:165]: Log initialized
[21:49:07][C][safe_mode:084]: Unsuccessful boot attempts: 9
[21:49:07]Guru Meditation Error: Core  0 panic'ed (StoreProhibited). Exception was unhandled.
[21:49:07]
[21:49:07]Core  0 register dump:
[21:49:07]PC      : 0x4200c133  PS      : 0x00060230  A0      : 0x82004ff2  A1      : 0x3fcaad70  
[21:49:07]A2      : 0x00000000  A3      : 0x0000ea60  A4      : 0x00000000  A5      : 0x3fc9d5a0  
[21:49:07]A6      : 0x3fca4890  A7      : 0x40000744  A8      : 0xa0ffff91  A9      : 0xa0ffff95  
[21:49:07]A10     : 0x3fcac330  A11     : 0x3fcaad70  A12     : 0x3fcaadc8  A13     : 0x0000000b  
[21:49:07]A14     : 0x0000000b  A15     : 0x0000000c  SAR     : 0x0000001d  EXCCAUSE: 0x0000001d  
[21:49:07]EXCVADDR: 0xa0ffff95  LBEG    : 0x40056f5c  LEND    : 0x40056f72  LCOUNT  : 0xffffffff  
[21:49:07]
[21:49:07]
[21:49:07]Backtrace: 0x4200c130:0x3fcaad70 0x42004fef:0x3fcaae10
[21:49:07]
[21:49:07]
[21:49:07]
[21:49:07]
[21:49:07]ELF file SHA256: f0a7effd8
[21:49:07]
[21:49:07]Rebooting...
```

### Additional information

_No response_

I already posted the back trace above and root caused this. 

If you add this it will catch the issue when you compile:
```
esphome:
  includes:
    - <rom/uart.h>
```

Log:
```
src/main.cpp:24:25: error: 'esphome::script::SingleScript<>* send_packet' redeclared as different kind of entity
   24 | script::SingleScript<> *send_packet;
      |                         ^~~~~~~~~~~
In file included from src/main.cpp:4:
/home/jon/.platformio/packages/framework-espidf/components/esp_rom/esp32s3/include/esp32s3/rom/uart.h:303:6: note: previous declaration 'void send_packet(uint8_t*, int)'
  303 | void send_packet(uint8_t *p, int len);
      |      ^~~~~~~~~~~
src/main.cpp: In function 'void setup()':
src/main.cpp:297:15: error: assignment of function 'void send_packet(uint8_t*, int)'
  297 |   send_packet = new script::SingleScript<>();
      |   ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/main.cpp:298:16: error: request for member 'set_name' in '* send_packet', which is of non-class type 'void(uint8_t*, int)' {aka 'void(unsigned char*, int)'}
  298 |   send_packet->set_name(LOG_STR("send_packet"));
      |                ^~~~~~~~
src/main.cpp:299:49: error: no matching function for call to 'esphome::Automation<>::Automation(void (&)(uint8_t*, int))'
  299 |   automation_id_2 = new Automation<>(send_packet);
      |                                                 ^
In file included from src/esphome/components/api/api_server.h:9,
                 from src/esphome/components/api/api_connection.h:8,
                 from src/esphome.h:3,
                 from src/main.cpp:3:
```

--- a/esphome/config_validation.py
+++ b/esphome/config_validation.py
@@ -256,8 +256,6 @@ RESERVED_IDS = [
     "one_bits",
     "recv_packet",
     "send_packet",
-    "sp",
-    "gpio_init",
     "check_pos",
     "software_reset",
 ]

--- a/esphome/config_validation.py
+++ b/esphome/config_validation.py
@@ -244,6 +244,22 @@ RESERVED_IDS = [
     "uart0",
     "uart1",
     "uart2",
+    # ESP32 ROM functions
+    "crc16_be",
+    "crc16_le",
+    "crc32_be",
+    "crc32_le",
+    "crc8_be",
+    "crc8_le",
+    "dbg_state",
+    "debug_timer",
+    "one_bits",
+    "recv_packet",
+    "send_packet",
+    "sp",
+    "gpio_init",
+    "check_pos",
+    "software_reset",
 ]
 
 

